crates/holochain/src/conductor/conductor.rs

```diff
@@ -837,12 +838,107 @@ mod network_impls {
+                let extrapolated_coverage =
+                    conn.p2p_extrapolated_coverage(agent_infos[0].storage_arc.inner().into())?;
+                let total_peers = (extrapolated_coverage[0] / arc_size) as u64;
+
+                // get sum of bytes from dht and cache db since last time
+                // request was made or since the beginning of time
+                let last_time_queried = match last_time_queried {
+                    Some(timestamp) => *timestamp,
+                    None => std::time::UNIX_EPOCH
+                        .elapsed()
+                        .map_err(|err| ConductorError::Other(Box::new(err)))?
+                        .as_millis(),
+                };
+                let sum_of_bytes_row_fn = |row: &Row| {
+                    row.get(0)
+                        .map(|maybe_bytes_received: Option<u64>| {
+                            maybe_bytes_received.unwrap_or_else(|| 0)
+                        })
+                        .map_err(|err| DatabaseError::SqliteError(err))
+                };
+                let dht_db = self
+                    .get_dht_db(dna)
+                    .map_err(|err| ConductorError::Other(Box::new(err)))?;
+                let dht_bytes_received = dht_db
+                    .async_reader({
+                        move |txn| {
+                            txn.query_row_and_then(
+                                SUM_OF_RECEIVED_BYTES_SINCE_TIMESTAMP,
+                                params![last_time_queried as u64],
+                                sum_of_bytes_row_fn,
+                            )
+                        }
+                    })
+                    .await?;
+
+                let cache_db = self
+                    .get_cache_db(&cell_id)
+                    .await
+                    .map_err(|err| ConductorError::Other(Box::new(err)))?;
+                let cache_bytes_received = cache_db
+                    .async_reader(move |txn| {
+                        txn.query_row_and_then(
+                            SUM_OF_RECEIVED_BYTES_SINCE_TIMESTAMP,
+                            params![last_time_queried as u64],
+                            sum_of_bytes_row_fn,
+                        )
+                    })
+                    .await?;
+                let bytes_since_last_time_queried = dht_bytes_received + cache_bytes_received;
+
+                // calculate open peer connections based on current gossip sessions
+                let completed_rounds_since_last_time_queried = diagnostics
+                    .metrics
+                    .read()
+                    .peer_node_histories()
+                    .iter()
+                    .flat_map(|(_, node_history)| node_history.completed_rounds.clone())
+                    .filter(|completed_round| {
+                        // this should compare to last_time_queried but I couldn't come up with that comparison
```

Comment:
aargh, I spent a long time trying to compare that Instant to milliseconds

Reply:
```rust
completed_round.start_time.elapsed().as_millis()
```

Reply:
Ah, i see you want this to be compared against an absolute timestamp... you're going to have to subtract out the `Instant::now()` and take it as a duration in the past to subtract out from the current epoch timestamp.

Reply:
My suggestion would be to get the system time first because that is a slower operation than getting an instant, so they'll be more tightly correlated if done in that order. And if you switch completed_round to a `Timestamp` as I suggest elsewhere, the code would become something like:

```rust
let now_abs = Timestamp::now();
let now_rel = tokio::time::Instant::now();
let diff = now_rel - completed_rount.start_time;
let start_abs = now_abs - diff;
start_abs > last_time_queried
```

Reply:
Instant and Timestamp also implement `PartialOrd`...so if you had two Timestamps couldn't you just compare them directly?

Reply:
That's what this code is doing, it's converting an instant into a timestamp.

Reply:
I think Michael meant this:
```rust
let now = tokio::time::Instant::now();
let round_start_time_diff = now - completed_round.start_time;
let round_start_timestamp =
    Timestamp::from_micros(round_start_time_diff.as_micros() as i64);
round_start_timestamp > last_time_queried
```